tests/integration/components/data-col-test.js

```diff
@@ -55,6 +56,29 @@ test('formats for multiple values', function(assert) {
   assert.equal(this.$('td')[0].innerText, 'Foo - Bar');
 });
 
+test('styles the cell with different width suffix', function(assert) {
+  set(this, 'item', { id: 1, name: 'Foo' });
+  set(this, 'property', {
+    key: 'name',
+    label: 'animal name',
+    style: {
+      width: '30px',
+      borderWidth: '2em',
+      borderStyle: 'dotted',
+      borderColor: 'blue'
+    }
+  });
+  this.render(hbs`{{data-col item=item property=property}}`);
+
+  let $cell = this.$('td').eq(0);
+
+  let em = parseInt($cell.css('font-size').replace('px', ''));
+  let borderWidth = 2 * em;
+
+  assert.equal($cell.css('width'),      '30px',             'width');
```

Comment:
"width" is not a correct name for this assert. You're testing wether you can use styling with different suffix, so use it in the name:
"styles cell for px value"
"styles cell for em value"